### PR TITLE
Adapt to trio norms + add @asynccontextmanager + add sphinx docs

### PR DESCRIPTION
**Pull request by njsmith**

### CODE REVIEW

**Comment by njsmith** [comment]
I cannot figure out why the coverage here is so messed up. We're using exactly the same scripts as trio. Yet for some reason it looks like the coverage gathering is only getting turned on *after* we import the module, so all the lines that are executed on import show up as being not-covered.

**Comment by njsmith** [comment]
AHHHH hahaha I figured it out.

We use `pytest-asyncio` as a convenient way to run async functions.

`pytest-asyncio` at some point... started using this library! So if pytest happens to load the `pytest-asyncio` plugin before it loads the `pytest-cov` plugin, then we get imported then too.

Also it's kind of bad if our testing infrastructure indirectly depends on the code-under-test, we should probably fix that anyway.

File: README.rst [hunk 1]
```diff
@@ -1,340 +1,121 @@
...
+   :target: https://gitter.im/python-trio/general
+   :alt: Join chatroom
+
+.. image:: https://img.shields.io/badge/docs-read%20now-blue.svg
+   :target: https://async-generator.readthedocs.io/en/latest/?badge=latest
+   :alt: Documentation Status
 
 .. image:: https://travis-ci.org/njsmith/async_generator.svg?branch=master
    :target: https://travis-ci.org/njsmith/async_generator
    :alt: Automated test status
```
Comment (pquentin) [inline README.rst]: Note that this image and the codecov one below link to njsmith/async_generator.

Comment (njsmith) [reply README.rst]: Whoops, good catch. Want to fix it? :-)

Comment (pquentin) [reply README.rst]: Sure, https://github.com/python-trio/async_generator/pull/10 :)

**Pull request merged**